skbio/metadata/_interval.py

```diff
@@ -0,0 +1,334 @@
+
+        Note
+        ----
+        There are two types of queries to perform
+        1. Query by interval.
+        2. Query by key/val pair (i.e. gene=sagA).
+
+        """
+        if self._is_stale_tree:
+            self._rebuild_tree(self._metadata)
+            self._is_stale_tree = False
+
+        invs = set()
+
+        # Find queries by interval
+        if intervals is not None:
+            for value in intervals:
+                invs.update(self._query_interval(value))
+
+        # Find queries by feature attribute
+        if len(invs) == 0 and metadata is not None:
+            invs = set(self._metadata)
+
+        if metadata is not None:
+            invs = self._query_attribute(invs, metadata)
+        return list(invs)
+
+    def drop(self, intervals=None, boundaries=None, metadata=None):
+        """ Drops Interval objects according to a specified query.
+
+        Drops all Interval objects that matches the query.
+
+        Parameters
+        ----------
+        intervals : iterable of tuple of ints
+            A list of intervals associated with the Interval object.
+        boundaries : iterable of tuple of bool
+            A list of boundaries associated with the Interval object.
+        metadata : dict
+            A dictionary of key word attributes associated with the
+            Interval object.
+
+        Examples
+        --------
+        >>> from skbio.metadata import IntervalMetadata
+        >>> interval_metadata = IntervalMetadata()
+        >>> interval_metadata.add(intervals=[(0, 2), (4, 7)],
+        ...                       boundaries=None, metadata={'name': 'sagA'})
+        >>> interval_metadata.add(intervals=[(40, 70)],
+        ...                       boundaries=None, metadata={'name': 'sagB'})
+        >>> interval_metadata.drop(metadata={'name': 'sagA'})
+        >>> interval_metadata.query(metadata={'name': 'sagA'})
+        []
+        """
+        if intervals is None:
+            intervals = []
+        if metadata is None:
+            metadata = {}
+
+        queried_invs = self.query(intervals=intervals,
+                                  boundaries=boundaries,
+                                  metadata=metadata)
+        new_invs = []
+        # iterate through queries and drop them
+        for inv in self._metadata:
+            if inv not in queried_invs:
+                new_invs.append(inv)
+        self._metadata = new_invs
+        self._is_stale_tree = True
+
+    def __eq__(self, other):
+        return sorted(self._metadata) == sorted(other._metadata)
+
+    def __ne__(self, other):
+        return not self.__eq__(other)
+
+    def __repr__(self):
+        if len(self._metadata) < 10:
+            return str(self._metadata)
+        else:
+            return str(self._metadata[:5] + ['...'] + self._metadata[-5:])
+
+
+def _polish_interval(interval):
+    if isinstance(interval, tuple):
+        if len(interval) == 0:
```

Comment:
I'm still seeing things not represented though when looking at the raw PR
(i.e., without a commit or message tag in the URL). For instance:

https://github.com/biocore/scikit-bio/pull/1356/files#r65108752

On Tue, May 31, 2016 at 10:39 AM, Jamie Morton <notifications@github.com>
wrote:

> In skbio/metadata/_interval.py
> <https://github.com/biocore/scikit-bio/pull/1356#discussion_r65228593>:
>
> > +    def __eq__(self, other):
> > +        return sorted(self._metadata) == sorted(other._metadata)
> > +
> > +    def __ne__(self, other):
> > +        return not self.__eq__(other)
> > +
> > +    def __repr__(self):
> > +        if len(self._metadata) < 10:
> > +            return str(self._metadata)
> > +        else:
> > +            return str(self._metadata[:5] + ['...'] + self._metadata[-5:])
> > +
> > +
> > +def _polish_interval(interval):
> > +    if isinstance(interval, tuple):
> > +        if len(interval) == 0:
>
> Everything I've done so far has actually been pushed. See this line
> <https://github.com/biocore/scikit-bio/pull/1356/files#diff-926d456b8487e3e1349844a7f421b28fR343>
>
> The problem here is the comments are still showing the old code ...
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/biocore/scikit-bio/pull/1356/files/82ffdeb392ae0f4104cc071cb5811cd92a802ed7#r65228593>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AAc8suDiDGgwc40UbrH9Wg_EfnfJDi44ks5qHHJEgaJpZM4Ig0qO>
> .
>